src/MongoDB.Driver.Core/Core/Configuration/ClusterSettings.cs

```diff
@@ -75,7 +85,21 @@ public class ClusterSettings
             Optional<IReadOnlyDictionary<string, BsonDocument>> schemaMap = default(Optional<IReadOnlyDictionary<string, BsonDocument>>),
             Optional<ConnectionStringScheme> scheme = default(Optional<ConnectionStringScheme>))
         {
+            _connectionModeSwitch = connectionModeSwitch.WithDefault(ConnectionModeSwitch.NotSet);
+#pragma warning disable CS0618
+            if (connectionMode.HasValue && !connectionModeSwitch.HasValue)
+            {
+                // the ClusterConnectionModeSwitch value has not been provided
+                throw new InvalidOperationException("ClusterSettings.ConnectionMode must be used only with ClusterConnectionModeSwitch.");
+            }
```

Comment:
Move error checking to call to helper method below (**after** assigning default values to fields).